### PR TITLE
Routing connection from popups to detail pages

### PR DESCRIPTION
**Pull request by vorugantia**
@cdeil - I'm opening this as a PR just so you can comment on the style of the buttons/links to the popup/detail page.

Here's what the popup looks like for gamma-cat sources:

popuo

I didn't do much styling, but how does the "More information" link look? Do you think it would look better as a button instead?

### CODE REVIEW

**Comment by vorugantia** [comment]
The URL now includes glon/glat coordinate parameters, and the Aladin map is initialized to read these coordinates from the URL. That means people can now share map views with each other!

This is starting to become a big PR... I don't think you have to review everything. If you want, you can just check how I moved the Aladin Lite "config" over from `map.component.ts` to `map-state.ts`.

File: src/app/widgets/cat-source/cat-source-tev/cat-source-tev.component.ts [hunk 1]
```diff
@@ -55,9 +55,28 @@ export class CatSourceTeVComponent implements OnInit {
     return false;
   }
 
+  goToMap() {
+    let formattedGlon = this.source.format(this.d.glon, false).toString();
```
Comment (cdeil) [inline src/app/widgets/cat-source/cat-source-tev/cat-source-tev.component.ts]: Maybe move all of this code into a property `aladin_target_string` on the source class?

File: src/app/widgets/popup/popup-3fgl.ts [hunk 2]
```diff
@@ -68,14 +51,25 @@ export class Popup3FGL {
...
-
+        ` + this.getSourceDetailUrl() + `
     `;
   }
 
   round(val) {
     return (Math.round(val * 100) / 100).toFixed(2);
   }
 
+  getSourceDetailUrl() {
```
Comment (cdeil) [inline src/app/widgets/popup/popup-3fgl.ts]: Maybe have a separate function that generates just the URL string.
Not real code, just roughly, the return value would be a string with the relative URL:
```
#cat/{source.cat_name}/{source.source_id}`
```
The rest of these functions, the HTML with `<div class='popup-link'>` is duplicated for each cat. This 4x code duplication could be easily avoided, no?

**Review by cdeil — COMMENTED** [review]
I left two inline comments.

Can't wait to try this out! I can't comment on the router code from looking at it here, just go ahead and merge whenever you like.

And I think it's nice how you removed the code duplication for the CSS!

File: src/app/widgets/cat-source/cat-source-3fhl/cat-source-3fhl.component.ts [hunk 3]
```diff
@@ -51,7 +51,9 @@ export class CatSource3FHLComponent implements OnInit {
   ngOnInit() {
     console.log("Routing to CatSource3FHLComponent...");
 
+    // TODO params will be replaced by paramMap  https://angular.io/guide/router#parammap-api
```
Comment (vorugantia) [inline src/app/widgets/cat-source/cat-source-3fhl/cat-source-3fhl.component.ts]: Reminder: do this TODO

**Comment by vorugantia** [comment]
Comments have been addressed. I finished linking the popup to the cat detail page, so I'm closing this PR. In another PR I can make the URL parameterization more advanced.

**Pull request merged**